### DISCUSSION TITLE
No Sound

### DISCUSSION THREAD

**Discussion by jd1500** [SpiralLinux/SpiralLinux-project#36] [Q&A] [answered]
New SpiralLinux 11.2 install, everything works great except there's  no sound at all.

 Zero output on right or left speaker on any app.  No system sounds, etc.  Acts like speakers are disconnected.
 Checked all wiring and connections,  Rebooted, etc.  Sound works fine on other Linux and win10 setups.

SpiralLinux is awesome otherwise, but no sound is a  deal breaker.

Please help!  Thanks for any replies.

**Comment by geckolinux** [comment, marked as answer]
Hi there, could you please give the details of which ISO version you used to install from?

**Reply by jd1500** [reply]
Thank you for your reply.  Some system sounds work:, but no audio files or speech at all.

 ISO  downloaded:

https://versaweb.dl.sourceforge.net/project/spirallinux/11.220628/SpiralLinux_Mate_11.220628_x86-64.iso

**Reply by geckolinux** [reply]
Hi there, sorry for the trouble. I just re-tested that ISO and audio in Firefox and VLC is working as expected.

Please check your hardware and speaker setup again. Also you could check the `pavucontrol` configurations to make sure that the correct audio device is selected for output. If it still doesn't work please try the live USB system again to see if audio works there.

**Reply by jd1500** [reply]
Thank you for your efforts, it's fixed now.  I apologize for the trouble and wasted time I've caused you.

I knew sooner or later you'd find out what an idiot I am.  For some reason all the audio on all my apps had the volume turned off after install.  It's not something I thought about, did not look at,  and so I thought it was broken... The truth is:  **SpiralLinux is not broken**...I am.    

Strangely enough the speaker test in "Sound Preferences" still not working.  So now, I can continue happily using Spiral...what a relief...I was not looking forward to reinstalling the old OS again. 

Thanks again for all you do, and I'm sure I'll further demonstrate what an idiot I am in the near future.

**Reply by geckolinux** [reply]
The important thing is that it's working correctly, thanks a lot for confirming that!

**Comment by NikkiSatmaka** [comment]
I also have an issue regarding sound. I don't know whether to open a new thread for this.

Basically, my system would not boot with sound.
However, after a restart, the sound would work just fine.

This happens on every first boot. A restart after the first boot, would make the sound work again.

What would be the source of the problem?

This is the content of `/proc/asound/cards`
```
0 [HDMI           ]: HDA-Intel - HDA Intel HDMI
                      HDA Intel HDMI at 0xf7c34000 irq 33
 1 [PCH            ]: HDA-Intel - HDA Intel PCH
                      HDA Intel PCH at 0xf7c30000 irq 34
 2 [Device         ]: USB-Audio - USB Audio Device
                      GeneralPlus USB Audio Device at usb-0000:00:14.0-1, full speed
 3 [webcam         ]: USB-Audio - Full HD webcam
                      SHENZHEN AONI ELECTRONIC CO., LTD Full HD webcam at usb-0000:00:14.0-13, high s
```

**Reply by geckolinux** [reply]
Hi there @NikkiSatmaka have you tried with both provided kernels (5.18 and 5.10)?

**Reply by NikkiSatmaka** [reply]
Yess, both have the same problem.
I'm assuming that something fails to load during a cold boot, but succeeded after a reboot

**Reply by geckolinux** [reply]
I see, thanks. Have you seen this issue with other distros as well?

**Reply by NikkiSatmaka** [reply]
I used arch before in 2020, but I never encountered this problem.
I just returned from windows and installed vanilla debian, then SpiralLinux shortly after

**Reply by geckolinux** [reply]
> I used arch before in 2020, but I never encountered this problem.

I assume with this same hardware?

Does this happen every other time when booting the SpiralLinux live system too, or just the installed system? If it also happens with the live system is there any way you could test a vanilla Debian live ISO to confirm if it behaves the same?

It's definitely possible for an audio device to work sporadically. It's technically a fault or "quirk" with the hardware, but as users we expect the kernel to work around those quirks. Here's a personal example that I ran into, which required a patch at the kernel level:
https://bugzilla.suse.com/show_bug.cgi?id=1172017

**Reply by NikkiSatmaka** [reply]
> I assume with this same hardware?

Yess, same hardware.

I also installed Spiral XFCE on my thinkpad X230, and this bug never occurs. So yeah, I guess it is a hardware "quirk".
The surprising thing is, today I booted my PC and the sound worked!! Without a reboot needed. This is weird. So the occurance is not as predictable as I thought.

> Does this happen every other time when booting the SpiralLinux live system too, or just the installed system? If it also happens with the live system is there any way you could test a vanilla Debian live ISO to confirm if it behaves the same?

I didn't check the sound on the live system. Wait a sec, I'll check the live ISO ASAP

**Reply by NikkiSatmaka** [reply]
Booted into the live ISO multiple times, and sound always worked.

My installed system is misbehaving differently now. The right speaker would output sound, but the left one wouldn't. A reboot does not fix this and I have a non functional left speaker. A boot into the live ISO, both speaker would work flawlessly

**Reply by bittin** [reply]
had a problem with my sound today too, when i used it for the first time, downgrading pulseaudio from 15 to 14.2 fixed it for me however

**Reply by NikkiSatmaka** [reply]
My pulseaudio is 14.2 from the start. So I guess it's not the source of the problem

**Reply by geckolinux** [reply]
> My installed system is misbehaving differently now. The right speaker would output sound, but the left one wouldn't. A reboot does not fix this and I have a non functional left speaker. A boot into the live ISO, both speaker would work flawlessly

Huh, that's a weird one. Can you try with the headphone port? Also check `alsamixer` to make sure that the left/right balance isn't skewed.
